src/services/dsnp.ts

```diff
@@ -0,0 +1,267 @@
+ * @return all DSNP registration (DSNP id and handle) for which the given address is authorized to sign.
+ */
+export const getSocialIdentities =
+  core.contracts.registry.getRegistrationsByWalletAddress;
+
+/**
+ * createNewDSNPRegistration creates a new DSNP identity for a given wallet address.
+ * This function will register a new DSNP Id with the provided handle and then delegate authority
+ * to managed the Id to the owner of the provided wallet address.
+ * @param addr address of an on-chain account
+ * @param handle handle to be registered with the new Id.
+ * @returns the DSNPUserURI for the new account.
+ * @throws if registration fails for any reason, or the handle is already registered.
+ */
+export const createNewDSNPRegistration = createRegistration;
+
+/**
+ * startSubscriptions starts listenting for DNSP publication events and DSNP registration change
+ * events. The subscriptions start with all previous events which will be immediately emitted to
+ * the handlers, and then will continue listening for new events until the returned unsubscribe
+ * function is called.
+ * @param batchHandler a function called when a batch announcement is made on chain.
+ * @param registryHandler a function called when a DSNP registration is made or changed.
+ * @returns an unsubscribe function that will stop the subscriptions when called.
+ */
+export const startSubscriptions = async (
+  batchHandler: BatchAnnouncementHandler,
+  registryHandler: RegistryUpdateHandler
+): Promise<UnsubscribeFunction> => {
+  let blockNumber: number;
+  let blockIndex = 0;
+
+  // subscribe to all announcements
+  const unsubscribeToBatchPublications = await core.contracts.subscription.subscribeToBatchPublications(
+    (announcement: BatchPublicationLogData) => {
+      if (announcement.blockNumber !== blockNumber) {
+        blockNumber = announcement.blockNumber;
+        blockIndex = 0;
+      } else {
+        blockIndex++;
+      }
+      batchHandler(announcement, blockIndex);
+    },
+    { fromBlock: 1 }
+  );
+
+  // subscribe to registry events
+  const unsubscribeToRegistryUpdate = await core.contracts.subscription.subscribeToRegistryUpdates(
+    registryHandler,
+    { fromBlock: 1 }
+  );
+
+  return () => {
+    unsubscribeToRegistryUpdate();
+    unsubscribeToBatchPublications();
+  };
+};
+
+/**
+ * Load a DSNP batch file from a url in a batch announcment, parse it, and emit all rows in the batch
+ * to the given handler.
+ * @param batchAnnouncement batch announcment that specifies where to retrieve the batch file.
+ * @param rowHandler handler function that receives all rows from the batch.
+ * @throwss if batch cannot be read.
+ */
+export const readBatchFile = async (
+  batchAnnouncement: BatchPublicationLogData,
+  rowHandler: AnnouncementRowHandler
+): Promise<void> => {
+  let batchIndex = 0;
+  const reader = await core.batch.openURL(
+    (batchAnnouncement.fileUrl.toString() as any) as URL
+  );
```

Comment:
Not yet, but once sdk v3.0.2 is out this can just be `batchAnnouncement.fileUrl`